main.py

```diff
@@ -1,6 +1,7 @@
 """Proxy server to communicate a client to ARTIQ."""
 
 import ast
+import h5py
```

Comment:
I think this is not a standard library. Please move it to the external library imports.
Also, please check out the GitHub action errors. Maybe `h5py` should be added to the dependency.